sdks/java/io/kafka/src/main/java/org/apache/beam/sdk/io/kafka/KafkaUnboundedReader.java

```diff
@@ -619,7 +620,7 @@ private void nextBatch() {
     partitionStates.forEach(p -> p.recordIter = records.records(p.topicPartition).iterator());
 
     // cycle through the partitions in order to interleave records from each.
-    curBatch = Iterators.cycle(new ArrayList<>(partitionStates));
+    curBatch = Iterators.cycle(new ArrayDeque<>(partitionStates));
```

Comment:
This list is neither used as a stack or nor as a queue. I would say you can either leave it as ArrayList or LinkedList. ArrayDeque incorrectly implies usage.